contracts/SmartLoansFactory.sol

```diff
@@ -36,36 +39,42 @@ contract SmartLoansFactory is IBorrowersRegistry {
     pool = _pool;
     priceProvider = _priceProvider;
     assetsExchange = _assetsExchange;
+    SmartLoan smartLoanImplementation = new SmartLoan();
+    upgradeableBeacon = new UpgradeableBeacon(address(smartLoanImplementation));
+    upgradeableBeacon.transferOwnership(msg.sender);
   }
 
   function createLoan() public returns(SmartLoan) {
-    SmartLoan newAccount = new SmartLoan(priceProvider, assetsExchange, pool);
+    BeaconProxy beaconProxy = new BeaconProxy(payable(address(upgradeableBeacon)), abi.encodeWithSelector(SmartLoan.initialize.selector, address(priceProvider), address(assetsExchange), address(pool)));
+    SmartLoan smartLoan = SmartLoan(payable(address(beaconProxy)));
 
     //Update registry and emit event
-    updateRegistry(newAccount);
-    newAccount.transferOwnership(msg.sender);
+    updateRegistry(smartLoan);
+    smartLoan.transferOwnership(msg.sender);
 
-    return newAccount;
+    return smartLoan;
   }
 
   function createAndFundLoan(uint256 _initialDebt) external payable returns(SmartLoan) {
-    SmartLoan newAccount = new SmartLoan(priceProvider, assetsExchange, pool);
+    BeaconProxy beaconProxy = new BeaconProxy(payable(address(upgradeableBeacon)), abi.encodeWithSelector(SmartLoan.initialize.selector, address(priceProvider), address(assetsExchange), address(pool)));
+    SmartLoan smartLoan = SmartLoan(payable(address(beaconProxy)));
+    smartLoan.initialize(priceProvider, assetsExchange, pool);
```

Comment:
Wasn't it already initialized in proxy creation 2 lines above?

Reply:
You're right.